nixos/modules/services/misc/matrix-dendrite.nix

```diff
@@ -0,0 +1,179 @@
+              Lists of domains that the server will trust as identity
+              servers to verify third
+            '';
+          };
+          disable_federation = lib.mkOption {
+            type = lib.types.bool;
+            default = true;
+            description = ''
+              Whether to enable federation with other matrix servers.
+            '';
+          };
+        };
+        options.client_api = {
+          registration_disabled = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              Whether to disable user registration to the server
+              without the shared secret.
+            '';
+          };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for dendrite, see:
+        <link xlink:href="https://github.com/matrix-org/dendrite"/>
+      '';
+    };
+  };
+
+  config = lib.mkMerge [
+    (lib.mkIf cfg.enable {
+      systemd.services.matrix-dendrite = {
+        description = "Dendrite Matrix homeserver";
+        after = [
+          "network.target"
+        ] ++ lib.optionals (cfg.generate_private_key == true) [
+          "matrix-dendrite-key-generator.service"
+        ];
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Type = "notify";
+          DynamicUser = true;
+          StateDirectory = cfg.stateDir;
+          WorkingDirectory = workingDir;
+          ExecStart = lib.strings.concatStringsSep " " ([
+            "${pkgs.matrix-dendrite}/bin/dendrite-monolith-server"
+            "--config ${configurationYaml}"
+            "--http-bind-address :${builtins.toString cfg.httpPort}"
+          ] ++ lib.optionals (cfg.httpsPort != null) [
+            "--https-bind-address :${builtins.toString cfg.httpsPort}"
+          ] ++ lib.optionals (cfg.tls_cert != null) [
+            "--tls-cert ${cfg.tls_cert}"
+          ] ++ lib.optionals (cfg.tls_key != null) [
+            "--tls-key ${cfg.tls_key}"
+          ]);
+          ExecReload = "${pkgs.util-linux}/bin/kill -HUP $MAINPID";
+          Restart = "on-failure";
+        };
+      };
+    }
+    )
+    (lib.mkIf
+      (cfg.enable &&
+        cfg.generate_private_key == true || cfg.tls_cert != null ||
+        cfg.tls_key != null)
+      {
+        systemd.services.matrix-dendrite-key-generator = {
+          description = "Dendrite Matrix homeserver - Key Generator";
+          wantedBy = [ "matrix-dendrite.service" ];
+          requiredBy = [ "matrix-dendrite.service" ];
+          before = [ "matrix-dendrite.service" ];
+          path = with pkgs; [ matrix-dendrite ];
+          serviceConfig = {
+            Type = "oneshot";
+            DynamicUser = true;
+            StateDirectory = cfg.stateDir;
+            WorkingDirectory = workingDir;
+            RemainAfterExit = true;
+          };
+          script = lib.strings.concatStringsSep " " ([
+            "generate-keys"
```

Comment:
I suggest doing this the same as the other service, so removing the `path = [ matrix-dendrite ]` and using
```nix
{
  serviceConfig.ExecStart = [ "${matrix-dendrite}/bin/generate-keys" ... ];
}
```
instead